File: LeetCodeTests/Solution1.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System.Linq;

namespace CSharp
{
    class Solution1
    {
        public int[] TwoSum(int[] nums, int target) {
            Dictionary<int, int> numDict = new Dictionary<int, int>();
            int length = nums.Length;
            for (int i=0; i < length; i++) {
                numDict.Add(nums[i], i);
            }

            for (int i=0; i < length; i++) {
                int remain = target - nums[i];
                if (numDict.ContainsKey(remain)){
                    int remain_index = numDict[remain];
                    if (i != remain_index) { 
                        return new int[] {0, 1};
                    }
                }
            }
            return new int[]{};
        }
        public void test_solution(){
            Console.WriteLine("test_solution1");
            int[] nums = {2,7,11,15};
            int[] ans = TwoSum(nums, 9);
            Assert.IsTrue(ans.SequenceEqual(new int[] {0, 1}));
        }
    }

}

File: LeetCodeTests/solution1.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) (ans [2]int) {
	numDict := make(map[int]int)
	length := len(nums)

	for i := 0; i < length; i++ {
		num_i := nums[i]
		numDict[num_i] = i
	}

	for i := 0; i < length; i++ {
		remain := target - nums[i]
		if remain_index, ok := numDict[remain]; ok {
			if i != remain_index {
				ans = [2]int{i, remain_index}
				return
			}
		}
	}
	return
}

func main() {
	r := twoSum([]int{3, 2, 4}, 6)
	fmt.Println(r)
}
